enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,347 @@
+
+
+
+### Goals
+
+- To provide a common user and developer experience when installing and developing across cloud platforms
+- To be backwards compatible and fully satisfy the requirements of install-config type APIs.
+- To keep the user experience for day-zero operations unchanged or improved.
+- To not require any new runtime dependencies.
+- To provide an extensible framework to plug-in new infrastructure cloud providers.
+- To maintain compatibility with hive, particularly regarding `destroy` functionality
+
+### Non-Goals / Future work
+
+- Non-goal: To maintain a consistent infrastructure footprint as clusters previously created with Terraform
+- Future work: To optimize build processes or binary size (this will be prioritized ASAP in a separate design doc)
+- Future work: To use an existing management cluster to install OpenShift
+- Future work: To pivot the CAPI manifests to the newly-installed cluster to enable day-2 infrastructure management within the cluster.
+
+## Proposal
+
+The Installer will create CAPI infrastructure manifests based on user
+input from the install config; then, in order to provision cluster infrastructure,
+apply the manifests to CAPI controllers running on a local Kubernetes control-plane
+setup by [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest). 
+
+### Workflow Description
+
+**cluster creator** is a human user responsible for deploying a
+cluster. Note that the workflow does not change for this user.
+
+**openshift-install** is the Installer binary.
+
+1. The cluster creator provides an install-config and credentials
+2. (optional) The cluster creator runs `openshift-install create manifests`
+3. (optional) The cluster creator edits the newly created CAPI manifests.
+4. The cluster creator runs `openshift-install create cluster`
+5. `openshift-install` extracts binaries for kube-apiserver, etcd, CAPI infrastructure provider & cloud CAPI provider to the install dir
+6. `openshift-install` using `envtest` initializes a control plane locally on the Installer host
+7. `openshift-install` execs the CAPI infrastructure and cloud provider as subprocesses, pointing them to the local control plane
+8. `openshift-install` applies the CAPI manifests to the control plane
+9. The CAPI controllers provision cluster infrastructure based on the manifests
+10. `openshift-install` monitors the status of the local manifests as they are applied
+11. If the statuses are as expected, infrastrucutre has been provisioned and installation continues with the normal flow.
+
+In the case of an error in the final step, the Installer will bubble up resources with non-expected statuses.
+
+#### Variation and form factor considerations [optional]
+
+
+### API Extensions
+
+As a result of this enhancement, API Extensions will only be used locally by the 
+local Installer control plane--they will not, until future work, be pivoted to the cluster.
+The following are CRD references for Installer-supported platforms:
```

Comment:
What about bare metal IPI? Would this be an opportunity to deprecate it? Or are we keeping it and embracing capi-metal3 for cluster creation?

Reply:
I have added a link to the metal3 capi provider (please LMK if you can find a better one). baremetal ipi, from my understanding, is still an open question (for @dtantsur & team). One of the primary motivations for this enhancement is to help lay out the design and see if it would be a good fit for baremetal ipi.